options/db_options.cc

```diff
@@ -268,12 +290,12 @@ std::unordered_map<std::string, OptionTypeInfo>
           OptionType::kUInt64T, OptionVerificationType::kNormal,
           OptionTypeFlags::kNone, 0}},
         {"access_hint_on_compaction_start",
-         {offsetof(struct DBOptions, access_hint_on_compaction_start),
-          OptionType::kAccessHint, OptionVerificationType::kNormal,
-          OptionTypeFlags::kNone, 0}},
-        {"info_log_level",
-         {offsetof(struct DBOptions, info_log_level), OptionType::kInfoLogLevel,
```

Comment:
Can you write a comment of instruction for developer who want to add new enum member or enum, where to change the code. Originally, it is straight forward: add the member, add the entry to the <string, optiontypeinfo> map with parameters for construction. But now, it has one more layer of OptionTypeInfo::Enum. May cause confusion.

Reply:
I updated the comment in OptionTypeInfo to describe how to create an OptionTypeInfo::Enum.

Note that the new procedure (write a map and use OptionTypeInfo::Enum is much simpler than the old process (write a map, add a OptionType, add an OptionTypeInfo to the options, add handling of that new OptionType to ParseOptionHelper, SerializeSingleOptionHelper, and AreOptionsEqual) -- 4 of the steps have been eliminated.